src/sentry/snuba/discover.py

```diff
@@ -145,16 +200,32 @@ def resolve_column(col):
     """
     if col is None:
         return col
+    elif isinstance(col, (list, tuple)):
+        return col
     # Whilst project_id is not part of the public schema we convert
     # the project.name field into project_id way before we get here.
     if col == "project_id":
         return col
     if col.startswith("tags[") or QUOTED_LITERAL_RE.match(col):
         return col
+
     return DISCOVER_COLUMN_MAP.get(col, u"tags[{}]".format(col))
 
 
-def resolve_discover_aliases(snuba_args):
+# TODO (evanh) Since we are assuming that all string values are columns,
+# this will get tricky if we every have complex columns where there are
+# string arguments to the functions that aren't columns
```

Comment:
Should we consider a better typed function representation that we can reason about better in sentry? Perhaps a lite version of an AST?

Reply:
I think that's where this is all heading ultimately. Since this code just translates API requests into snuba queries, we'll have to expand this logic out to match what snuba does. I would like to eventually spend time really seeing if we can rewrite some of the discover/event search code to make some of this logic clearer.

Reply:
> Since this code just translates API requests into snuba queries, we'll have to expand this logic out to match what snuba does. I would like to eventually spend time really seeing if we can rewrite some of the discover/event search code to make some of this logic clearer.

I think sentry can have a higher level model for functions, fields and literal parameters though. Ideally once snuba's AST is farther along we can push much of these custom functions into snuba query processors.

Reply:
Yeah, I think this is a good idea.